commands/viewport.go

```diff
@@ -35,8 +37,55 @@ func (cmd *Viewport) Parse() error {
 	switch lit {
 	case "down":
 		cmd.relative = 1
+		cmd.movecursor = false
 	case "up":
 		cmd.relative = -1
+		cmd.movecursor = false
+	case "halfpgdn", "halfpagedn", "halfpagedown":
```

Comment:
The functionality in these four switch cases seem pretty similar. Can you split them out into one or two parametrized functions, to keep the Parse function in low complexity?

Reply:
Done this.